packages/kotti-ui/source/make-props.test.ts

```diff
@@ -1,51 +1,143 @@
 import { PropOptions } from '@vue/composition-api'
+import { castArray, isEqual } from 'lodash'
+import { PropType } from 'vue'
 import { z } from 'zod'
 
 import { makeProps } from './make-props'
 import { silenceConsole } from './test-utils/silence-console'
 
-const expectDefault = (
-	prop: PropOptions<unknown, boolean>,
-	defaultValue: unknown,
-) => {
-	// @ts-expect-error
-	expect(prop.default()).toEqual(defaultValue)
-	expect(prop.required).toBe(undefined)
-}
-
-const expectRequired = (prop: PropOptions<unknown, boolean>) => {
-	expect(prop.default).toEqual(undefined)
-	expect(prop.required).toBe(true)
-}
-
-const expectValidator = (
-	prop: PropOptions<unknown, boolean>,
-	successCases: unknown[],
-	failCases: unknown[],
-) => {
-	const { validator } = prop
-
-	if (validator === undefined) {
-		expect(validator).not.toBe(undefined)
-		throw new Error('unexpected undefined validator')
+declare global {
+	namespace jest {
+		interface Matchers<R> {
+			toBeRequired(): R
+			toBeType(expectedPropType: PropType<unknown>): R
+			toDefaultTo(defaultValue: unknown): R
+			toValidate(...cases: unknown[]): R
+		}
 	}
-
-	successCases.forEach((c) => expect(validator(c)).toBe(true))
-	failCases.forEach((c) => expect(validator(c)).toBe(false))
 }
 
+expect.extend({
+	toBeRequired(prop: PropOptions<unknown, boolean>) {
+		const passNoDefault = prop.default === undefined
+		const passRequired = prop.required === true
+
+		return {
+			message: () => {
+				if (!passNoDefault)
+					return `Expected “prop.default” to be “undefined”. Was “${prop.default}”`
+				if (!passRequired)
+					return `Expected “prop.required” to be “true”. Was “${prop.required}”`
+				return 'valid'
+			},
+			pass: passNoDefault && passRequired,
+		}
+	},
+	toBeType(prop: PropOptions<unknown, boolean>, expected: PropType<unknown>) {
+		const actualPropTypes = new Set(castArray(prop.type as PropType<unknown>))
+		const expectedPropTypes = new Set(castArray(expected))
+
+		const pass = isEqual(actualPropTypes, expectedPropTypes)
+
+		return {
+			message: () =>
+				pass
+					? 'valid'
+					: this.utils.printDiffOrStringify(
+							expectedPropTypes,
+							actualPropTypes,
+							'Expected prop.type',
+							'Received prop.type',
+							this.expand,
+					  ),
+			pass,
+		}
+	},
+	toDefaultTo(prop: PropOptions<unknown, boolean>, expectedDefault: unknown) {
+		if (typeof prop.default !== 'function')
+			return {
+				message: () => 'Expected “prop.default” to be a function',
+				pass: false,
+			}
+
+		const actualDefaultValue = prop.default()
+		const passDefault = isEqual(actualDefaultValue, expectedDefault)
```

Comment:
minor: for consistency either rename `actualDefaultValue` to `actualDefault` or the argument to `expectedDefaultValue`

I prefer the latter but it's up to you